### PR TITLE
Implement structure data to alert,program & homepage

### PR DESCRIPTION
**Pull request by theasteve**

### CODE REVIEW

File: wp-content/themes/access/views/programs/single.twig [hunk 1]
```diff
@@ -30,7 +30,7 @@
       </div>
 
       <div>
-        <h1 class="c-header__descriptor" itemprop="alternateName">
+        <h1 class="c-header__descriptor" itemprop="name">
```
Comment (devowhippit) [inline wp-content/themes/access/views/programs/single.twig:33]: Undo this change

File: wp-content/themes/access/views/programs/single.twig [hunk 2]
```diff
@@ -53,9 +53,16 @@
           </span>
           {% endif %}
         </h2>
+        <p itemprop="datePosted" class="color__alt mt-1">
```
Comment (devowhippit) [inline wp-content/themes/access/views/programs/single.twig:56]: Add attribute to a hidden meta with ISO formated date

File: wp-content/themes/access/views/home.twig [hunk 3]
```diff
@@ -55,26 +55,31 @@
 
             <div class="screen-tablet:layout-columns-gutter">
               {% for tout in homepage_touts %}
-              <div class="c-card p-0 pb-4">
+              <div itemscope itemprop="SpecialAnnouncement" itemtype="https://schema.org/SpecialAnnouncement" class="c-card p-0 pb-4">
```
Comment (devowhippit) [inline wp-content/themes/access/views/home.twig:58]: Conditionally apply schema based on the COVID Status

File: wp-content/themes/access/controllers/programs.php [hunk 4]
```diff
@@ -146,6 +151,22 @@ public function getIconClass() {
...
   }
 
+  /**
+   * Get the itemtype for structure data
+   *
+   * @return  String  The itemtype
+   */
+
+  public function getItemScope() {
+    if ($this->status == 'covid-response') {
```
Comment (devowhippit) [inline wp-content/themes/access/controllers/programs.php:161]: @theasteve check to see if this works. `status` is either an array or false, never a string.

I think it would be `$this->status['type']`.

File: wp-content/themes/access/controllers/homepage-tout.php [hunk 5]
```diff
@@ -69,4 +74,20 @@ public function getStatus() {
...
+   * Get the itemtype for structure data
+   *
+   * @return  String  The itemtype
+   */
+
+  public function getItemScope() {
+    if ($this->custom['tout_status_type'] == 'covid-response') {
+      $item_scope = 'SpecialAnnouncement';
+    } else {
+      $item_scope = 'GovernmentService';
```
Comment (devowhippit) [inline wp-content/themes/access/controllers/homepage-tout.php:88]: Homepage Touts don't have a fallback schema. Gov't service only applies to programs.

File: wp-content/themes/access/views/components/alert.twig [hunk 6]
```diff
@@ -1,11 +1,20 @@
...
+
   <div class="c-alert-box__icon">
     <svg aria-hidden="true" class="c-alert-box__svg icon icon-{{ this.alert_icon }}">
       <use xlink:href="#icon-{{ this.alert_icon }}"></use>
     </svg>
   </div>
 
-  <div class="c-alert-box__body">
-    {{ this.alert_content }}
+  <div {% if this.itemprop_body %}itemprop="{{ this.itemprop_body }}" {% endif%}class="c-alert-box__body">
```
Comment (devowhippit) [inline wp-content/themes/access/views/components/alert.twig:13]: Instead of checking for the presence of `this.itemprop_body` I think it should check `this.item_scope`.

File: wp-content/themes/access/views/components/alert.twig [hunk 7]
```diff
@@ -1,11 +1,20 @@
...
+      {{ this.alert_content }}
   </div>
+
+  {% if this.item_scope %}
+    <meta itemprop="category" content="https://www.wikidata.org/wiki/Q81068910" />
```
Comment (devowhippit) [inline wp-content/themes/access/views/components/alert.twig:18]: This category is specific to COVID, so it should also only be present if the item scope status also equals `covid-response`. this may be best done in the controller.

File: wp-content/themes/access/views/home.twig [hunk 8]
```diff
@@ -55,26 +55,35 @@
 
             <div class="screen-tablet:layout-columns-gutter">
               {% for tout in homepage_touts %}
-              <div class="c-card p-0 pb-4">
+              <div itemscope itemprop="{{ tout.item_scope }}" itemtype="https://schema.org/{{tout.item_scope}}" class="c-card p-0 pb-4">
```
Comment (devowhippit) [inline wp-content/themes/access/views/home.twig:58]: These touts may not have structured data, so this needs to be conditional

File: wp-content/themes/access/views/home.twig [hunk 9]
```diff
@@ -55,26 +55,35 @@
...
                 <div class="c-card__body">
                   {% if tout.tout_title %}
-                  <h3 class="c-card__title mt-0">
+                  <h3 itemprop="name" class="c-card__title mt-0">
```
Comment (devowhippit) [inline wp-content/themes/access/views/home.twig:61]: Make conditional upon schema

File: wp-content/themes/access/views/home.twig [hunk 10]
```diff
@@ -55,26 +55,35 @@
...
                     {% if tout.status %}
                     <mark class="badge color-{{ tout.status.type }}-status">{{ tout.status.text }}</mark>
                     {% endif %}
 
-                    <a class="text-blue-dark" href="{{ tout.link_url }}"{% if tout.link_new_window %} target="_blank" rel="noopener noreferrer"{% endif %}>
+                    <a itemprop="newsUpdatesAndGuidelines" class="text-blue-dark" href="{{ tout.link_url }}"{% if tout.link_new_window %} target="_blank" rel="noopener noreferrer"{% endif %}>
```
Comment (devowhippit) [inline wp-content/themes/access/views/home.twig:66]: Make conditional upon schema

File: wp-content/themes/access/views/home.twig [hunk 11]
```diff
@@ -55,26 +55,35 @@
...
                       {{ tout.tout_title }}
                     </a>
                   </h3>
                   {% endif %}
 
                   <p class='c-card__subtitle color__alt'>
                     {{ __('Updated', 'accessnyctheme') }} {{ tout.post_modified|date('l, F jS, g:ia') }}
+                    <meta itemprop="datePosted" content="{{ tout.post_modified }}" />
```
Comment (devowhippit) [inline wp-content/themes/access/views/home.twig:74]: Make conditional upon the presence of itemscope

File: wp-content/themes/access/views/programs/single.twig [hunk 12]
```diff
@@ -30,7 +30,7 @@
       </div>
 
       <div>
-        <h1 class="c-header__descriptor" itemprop="alternateName">
+        <h1 class="c-header__descriptor" itemprop="alterName">
```
Comment (devowhippit) [inline wp-content/themes/access/views/programs/single.twig:33]: Is this right? `alterName` instead of `alternateName`?

File: wp-content/themes/access/views/programs/single.twig [hunk 13]
```diff
@@ -127,6 +135,24 @@
       </div>
     </section>
 
+    <div itemprop="governmentBenefitsInfo" itemtype="http://schema.org/GovernmentService" itemscope>
```
Comment (devowhippit) [inline wp-content/themes/access/views/programs/single.twig:138]: This block should be conditionally present. Only when the item scope is "specialannouncement" or the program status is "covid-response"

File: wp-content/themes/access/views/programs/single.twig (hunk 12 above)
Comment (theasteve) [reply wp-content/themes/access/views/programs/single.twig:33]: @devowhippit I think for this one is `alternameName` https://schema.org/alternateName

Comment (devowhippit) [reply wp-content/themes/access/views/programs/single.twig:33]: Ok. The change looks misspelled so make sure it's correct. Thanks!

File: wp-content/themes/access/views/components/alert.twig (hunk 7 above)
Comment (theasteve) [reply wp-content/themes/access/views/components/alert.twig:18]: @devowhippit does alert have fallback schema?
`this.item_scope` will either return `false` or `SpecialAnnouncement`, we can improve readability by `{% if this.item_scope  ==  SpecialAnnouncement %}` or `{% this.status['type'] == 'covid-response'  %}` Let me know if you agree with this implementation or if we need to add a fallback  schema.

File: wp-content/themes/access/views/home.twig [hunk 14]
```diff
@@ -55,34 +55,36 @@
...
                   {% endif %}
 
-                  {% if tout.item_scope == 'SpecialAnnouncement' %}
+                  {% if tout.item_scope %}
                     <meta itemprop="category" content="https://www.wikidata.org/wiki/Q81068910" />
                   {% endif %}
 
-                  {% if tout.custom.tout_status_clear_date %}
+                  {% if tout.custom.tout_status_clear_date and tout.item_scope %}
                     <meta itemprop="expires" content="{{ tout.custom.tout_status_clear_date }}" />
```
Comment (theasteve) [inline wp-content/themes/access/views/home.twig:88]: @devowhippit here I added the conditional `and tout.item_scope` to make sure the ` tout.custom.tout_status_clear_date` is not nil and is included. I can just make it `{% if tout.item_scope %}` if you like.

File: wp-content/themes/access/views/components/alert.twig (hunk 7 above)
Comment (devowhippit) [reply wp-content/themes/access/views/components/alert.twig:18]: @theasteve I am not certain if there is an applicable fallback. There may be something on this list but I think we can scope the task to just the SpecialAnnouncement schema https://schema.org/docs/full.html I think your readability suggestion is good!

File: wp-content/themes/access/views/components/card.twig [hunk 15]
```diff
@@ -24,14 +24,18 @@
...
 
       <p class="hide-for-print">
-        <a href="{{ post.get_permalink }}" {% if target_blank %}target="_blank" rel="noopener"{% endif %}>{{__('Learn more', 'accessnyctheme')}} <span class="sr-only">: {{ post.program_name }}</span></a>
+        <a itemprop="url" href="{{ post.get_permalink }}" {% if target_blank %}target="_blank" rel="noopener"{% endif %}>{{__('Learn more', 'accessnyctheme')}} <span class="sr-only">: {{ post.program_name }}</span></a>
       </p>
 
       {% if show_apply %}
       <p class="hide-for-print">
-        <a class="btn btn-secondary btn-next" href="{{ post.get_permalink }}{% if "how-to-apply" in post.show_sections %}?step=how-to-apply{% endif %}" {% if target_blank %}target="_blank" rel="noopener"{% endif %}>{{__('Apply', 'accessnyc-results')}}</a>
+        <a itemprop="url" class="btn btn-secondary btn-next" href="{{ post.get_permalink }}{% if "how-to-apply" in post.show_sections %}?step=how-to-apply{% endif %}" {% if target_blank %}target="_blank" rel="noopener"{% endif %}>{{__('Apply', 'accessnyc-results')}}</a>
```
Comment (devowhippit) [inline wp-content/themes/access/views/components/card.twig:32]: @theasteve I think there should only be one URL in the card so remove this one.
